test/gtest/mha_helper.hpp

```diff
@@ -399,14 +399,33 @@ void MultiHeadAttentionfp8(const tensor<T>& q_val,
 
     tensor<float> atten_heads_fp32(multi_head_attention_fp8.desc.GetLengths());
     // 2) fp8 matrix multiplication
-    Dot_4D_4D(softmax_fp8, v_val, atten_heads_fp32);
+    // check if q[N, H, S, D] and v[N, H, S, D]
+    //       or q[N, H, S, D] and v[N, H, D, S]
+    auto q_lens = q_val.desc.GetLengths();
+    auto v_lens = v_val.desc.GetLengths();
+    if(q_lens[2] == v_lens[2])
+    {
+        Dot_4D_4D(softmax_fp8, v_val, atten_heads_fp32);
+    }
+    else
+    {
+        Dot_4D_4D_T(softmax_fp8, v_val, atten_heads_fp32);
+    }
 
     // bring it back to fp32
     ScaleMult(atten_heads_fp32, s_descale * v_descale, atten_heads_fp32);
     aMax_O = AbsoluteMax(atten_heads_fp32);
 
     // scale to fp8 version
-    ScaleMult(atten_heads_fp32, o_scale, multi_head_attention_fp8);
+    if(q_lens[2] == v_lens[2])
+    {
+        ScaleMult(atten_heads_fp32, o_scale, multi_head_attention_fp8);
+    }
+    else
+    {
+        // ck solver
+        ScaleMult(atten_heads_fp32, GetF8Scaling(aMax_O), multi_head_attention_fp8);
```

Comment:
`aMax_O` must not be used here, it's an output parameter and will be passed as `scale_o` into the next iteration.